core/src/main/scala/Client.scala

```diff
@@ -346,9 +340,19 @@ object Client {
     protocol: String = HTTP,
     maxConnections: Int = 10
   )(implicit executor: ExecutionContext): Client = {
-    val (host0, port0, scheme0, ssl0, protocol0, maxConnections0, executor0) = (
-      host, port, scheme, ssl, protocol, maxConnections, executor
+    val (host0, port0, scheme0, protocol0, maxConnections0, executor0) = (
+      host, port, scheme, protocol, maxConnections, executor
     )
+    // Create a TLS configuration with SNI properly setup if no specific one provided.
+    val ssl0 = if(ssl == SSL.ClientConfiguration.default){
+      val ssl = new SSL.ClientConfiguration({
+        val sslContext = SSLContext.getInstance("TLS")
+        sslContext.init(null, null, null)
+        sslContext
+        }, "tls", host)
+        ssl
+      } else ssl
```

Comment:
I am sorry I am a bit nitpicking but:
- the indentation isn't great (also you might not have to create a ssl val if you return it immediately)
- having a ssl val that is hidden by the other one declared in the if statement is confusing 
Can you fix this ? Thx :)

Reply:
Indeed I don't need this intermediate variable, it's a leftover from my different tests. I removed it and fixed indentation.